### PR TITLE
Feature/data 2557

### PR DESCRIPTION
**Pull request by skavierin**

### CODE REVIEW

File: tests/unit/test_init.py [hunk 1]
```diff
@@ -86,3 +90,55 @@ def test_analyze_does_all_but_run(replica, create_relation):
...
         replica.assert_called_once()
         create_relation.assert_not_called()
+
+
+@patch('snowshu.core.main.ReplicaFactory.create')
+def test_custom_retry_count_cli_input(create):
+    # test if CLI input is passed to replica.create
+    runner = CliRunner()
+    runner.invoke(main.cli, ('create'))
+    create.assert_called_with(name=ANY, barf=ANY, retry_count=1)
```
Comment (skavierin) [inline tests/unit/test_init.py:100]: This bit passes locally but fails in CI, needs investigation

Comment (ArtemHU) [reply tests/unit/test_init.py:100]: I tested it locally - it fails
image

Could you please run all unit test cases from docker?
Also, please check the version of pytest

Comment (skavierin) [reply tests/unit/test_init.py:100]: Ran as it is supposed to be done 
image

Comment (ArtemHU) [reply tests/unit/test_init.py:100]: I reason why it fails - you have no replica.yml file in your root folder
By default it should reside in the root folder

Comment (ArtemHU) [reply tests/unit/test_init.py:100]: it looks like you need to supply the replica.yml file

Comment (skavierin) [reply tests/unit/test_init.py:101]: Guess it's fixed, recent run succeeded

File: snowshu/core/replica/replica_factory.py [hunk 2]
```diff
@@ -68,6 +73,7 @@ def _execute(self,
                                  self.config.source_profile.adapter,
                                  self.config.target_profile.adapter,
                                  threads=self.config.threads,
+                                 retry_count=self.retry_count,
```
Comment (ArtemHU) [inline snowshu/core/replica/replica_factory.py:76]: should we add it to cli?

Comment (eculicny) [reply snowshu/core/replica/replica_factory.py:76]: Let's add it in to the `analyze` command for consistency between available options.

**Review by eculicny — COMMENTED** [review]
Looks good! Only need to have the extra argument propagated to the `test_analyze.py` integration test.

**Comment by skavierin** [comment]
> Looks good! Only need to have the extra argument propagated to the test_analyze.py integration test.

Done

**Review by ArtemHU — APPROVED** [review]

**Pull request merged**